cargo-dist/templates/installer/installer.sh.j2

```diff
@@ -34,6 +34,49 @@ RECEIPT_HOME="${HOME}/.config/{{ app_name }}"
 BUILDER_GLIBC_MAJOR="2"
 BUILDER_GLIBC_SERIES="31"
 
+# We assign this right at the top of the file as a global
+# because a number of different things will try to read it.
+INSTALL_PATH_STRATEGY=""
+INSTALL_PATH_ENV_KEY=""
+INSTALL_PATH_SUBDIR=""
+INSTALL_PATH_HINT=""
+
+# If you're reading this install script generated by an app, and
+# this looks strange to you, it's because it's possible for there
+# to be multiple generated paths here. If you're only seeing one
+# $install_path assignment, it's because the app you're looking at
+# has been configured with a single install path.
+{% for install_path in install_paths %}
+if [ -z "$INSTALL_PATH_STRATEGY" ]; then
+{% if install_path.kind == "CargoHome" %}
+    # This strategy can't fail; always assign it
+    INSTALL_PATH_STRATEGY="CargoHome"
+    INSTALL_PATH_HINT="\$CARGO_HOME/bin (\$HOME/.cargo/bin)"
+{% elif install_path.kind == "HomeSubdir" %}
+    # We skip this if we can't determine the value of $HOME
+    if [ -n "${HOME:-}" ]; then
+        INSTALL_PATH_STRATEGY="HomeSubdir"
+        INSTALL_PATH_SUBDIR="{{ install_path.subdir }}"
+        INSTALL_PATH_HINT="\$HOME/$INSTALL_PATH_SUBDIR"
+    fi
+{% elif install_path.kind == "EnvSubdir" %}
+    # We skip this if the environment variable isn't set
+    if [ -n "{{ "${" }}{{ install_path.env_key }}:-}" ]; then
+        INSTALL_PATH_STRATEGY="EnvSubdir"
+        INSTALL_PATH_ENV_KEY="${{ install_path.env_key }}"
+        INSTALL_PATH_SUBDIR="{{ install_path.subdir }}"
+        INSTALL_PATH_HINT="\$$INSTALL_PATH_ENV_KEY/$INSTALL_PATH_SUBDIR"
+    fi
+{% else %}
+    {{ error("unimplemented install_path format: " ~ install_path.kind) }}
+{% endif %}
+fi
+{% endfor %}
+
+if [ -z "$INSTALL_PATH_STRATEGY" ]; then
+    err "could not find a valid path to install to; please check the installation instructions"
```

Comment:
oh yeah this would also be especially cruel with dynamic help text, because trying to get the installation instructions would just get this message again ☠️ 

(we should factor out the subroutine of usage() that prints the options and show it here? or maybe just fully invoke usage()?)

Reply:
By installation instructions I meant to refer people to the homepage, since our own usage text doesn't provide *that* much beyond "where would this be installed to", but we could still just reprint the usage text.